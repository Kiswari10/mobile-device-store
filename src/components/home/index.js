export * from './ProductCard';
export * from './SearchProduct';
export * from './SkeletonLoader';
