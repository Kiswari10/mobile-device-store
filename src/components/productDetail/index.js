export * from './ImageSection';
export * from './DescriptionSection';
export * from './ActionSection';
export * from './SkeletonLoaderDetail';
