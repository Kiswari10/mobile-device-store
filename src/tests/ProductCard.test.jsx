import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { ProductCard } from '../components';

describe('Pruebas en <ProductCard />', () => {
  const mock = {
    id: '1',
    imgUrl:
      'https://http2.mlstatic.com/D_NQ_NP_769912-MLA46490092919_062021-O.jpg',
    brand: 'Xiaomi',
    model: 'Redmi Note 10S',
    price: '340',
  };

  test('debe hacer match con el snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <ProductCard item={mock} key={mock.id} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('debe mostrar la imagen con la URL y el ALT indicado', () => {
    render(
      <MemoryRouter>
        <ProductCard item={mock} key={mock.id} />
      </MemoryRouter>
    );

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(mock.imgUrl);
    expect(alt).toBe(mock.model);
  });
});
