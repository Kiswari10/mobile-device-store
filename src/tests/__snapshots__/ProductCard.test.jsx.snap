// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductCard /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-hoverable css-dev-only-do-not-override-acm2ia"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="home_card-container-img"
      >
        <img
          alt="Redmi Note 10S"
          class="home_card-image"
          src="https://http2.mlstatic.com/D_NQ_NP_769912-MLA46490092919_062021-O.jpg"
        />
      </div>
      <div
        class="ant-card-meta"
      >
        <div
          class="ant-card-meta-detail"
        >
          <div
            class="ant-card-meta-title"
          >
            Xiaomi
          </div>
          <div
            class="ant-card-meta-description"
          >
            Redmi Note 10S
          </div>
        </div>
      </div>
      <p
        class="home_card-footer"
      >
        <span
          class="home_card-price"
        >
          Precio: $
          340
        </span>
        <a
          href="/product-detail/1"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-acm2ia ant-btn-round ant-btn-primary"
            type="button"
          >
            <span>
              Lo quiero
            </span>
          </button>
        </a>
      </p>
    </div>
  </div>
</div>
`;
